nexus/src/nexus.rs

```diff
@@ -1377,16 +1428,36 @@ impl Nexus {
             }
         }
 
-        self.disk_set_runtime(
-            opctx,
-            &authz_disk,
-            &db_disk,
-            self.instance_sled(&db_instance).await?,
-            sled_agent_client::types::DiskStateRequested::Attached(
-                *instance_id,
-            ),
-        )
-        .await?;
+        match &db_instance.runtime_state.state.state() {
+            // If there's a propolis zone for this instance, ask the Sled Agent
+            // to hot-plug the disk.
+            //
+            // TODO this will probably involve volume construction requests as
+            // well!
```

Comment:
Is this remarking that the API to `disk_set_runtime` is insufficient? I know most of this hot-plug path isn't implemented (which is 100% fine) but I just wanna make sure we're very explicit about what we expect to work / not work.

Reply:
propolis needs to be handed the volume construction request, so sled agent's disk_put is currently insufficient, yeah, but what do you think about

> We may want to remove the disk_put path and do everything through instance_put. That way we can funnel changes through instance_set_runtime and let propolis handle the disk attach or remove as it needs to.

?

Reply:
That change sounds good to me!